api/middleware/globalThrottler.go

```diff
@@ -0,0 +1,34 @@
+package middleware
+
+import (
+	"net/http"
+
+	"github.com/gin-gonic/gin"
+)
+
+// globalThrottler is a middleware global limiter used to limit total number of simultaneous requests
+type globalThrottler struct {
+	queue chan struct{}
+}
+
+// NewGlobalThrottler creates a new instance of a globalThrottler
+func NewGlobalThrottler(maxConnections uint32) *globalThrottler {
+	return &globalThrottler{
+		queue: make(chan struct{}, maxConnections),
```

Comment:
I think maxConnections should be at least 1.

Reply:
right, this might lead to a deadlock. Good catch! :+1:

Reply:
done